consensus/core/src/authority_node.rs

```diff
@@ -381,14 +383,35 @@ impl<C: CoreThreadDispatcher> NetworkService for AuthorityService<C> {
         // For now ask dag state directly
         let blocks = self.dag_state.read().get_blocks(&block_refs);
 
-        // Return the serialised blocks
+        // Now check if an ancestor's round is higher than the one that the peer has. If yes, then serve
+        // that ancestor blocks up to 10
```

Comment:
For sure we can do, but I wanted to limit the number here to avoid unnecessary duplication from potential inflight requests/processing. From some experimentation that I did capping the number of blocks is reducing the number of messages that need to get processed which didn't necessarily lead to better results.